server/matcher/match.go

```diff
@@ -152,10 +152,24 @@ func (m *Matcher) Match(book Booker, queue []*OrderRevealed) (seed []byte, match
 
 	updates = new(OrdersUpdated)
 	stats = new(MatchCycleStats)
-	startRate := midGap(book)
-	stats.StartRate = startRate
-	stats.LowRate = startRate
-	stats.HighRate = startRate
+
+	defer func() {
```

Comment:
I don't have a problem with this `defer`, but why is it a `defer`?  I see only one return this method.